lib/libvmod_std/vmod_std.c

```diff
@@ -311,3 +311,51 @@ vmod_fnmatch(VRT_CTX, VCL_STRING pattern, VCL_STRING subject,
 		flags |= FNM_PERIOD;
 	return (fnmatch(pattern, subject, flags) != FNM_NOMATCH);
 }
+
+static const void * const priv_task_id_ban = &priv_task_id_ban;
+
+VCL_BOOL v_matchproto_(td_std_ban)
+vmod_ban(VRT_CTX, VCL_STRING s)
+{
+	struct vmod_priv *priv_task;
+	VCL_STRING r;
+
+	CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);
+
+	priv_task = VRT_priv_task(ctx, priv_task_id_ban);
+	if (priv_task == NULL) {
+		VRT_fail(ctx, "std.ban(): no priv_task (out of workspace?)");
+		return (0);
+	}
+
+	r = VRT_ban_string(ctx, s);
```

Comment:
I would only bother grabbing a `priv_task` on error. That means zero cost error tracking when `std.ban_error()` is only used when `std.ban()` failed.

Reply:
good point

Reply:
actually, it is a good idea, but not possible at the moment:
For more than one `std.ban()` call, we need to reset any previous error. Yet to do so, we need to get a priv_task every time.
As I have already missed it in the past, I will suggest a "lookup priv_task without inserting".

Reply:
-> #3487